hudi-cli/src/main/java/org/apache/hudi/cli/commands/CommitsCommand.java

```diff
@@ -431,4 +429,23 @@ public String syncCommits(@CliOption(key = {"path"}, help = "Path of the table t
     return "Load sync state between " + HoodieCLI.getTableMetaClient().getTableConfig().getTableName() + " and "
         + HoodieCLI.syncTableMetadata.getTableConfig().getTableName();
   }
+
+  /*
+  Checks whether a commit or replacecommit action exists in the timeline.
+  * */
+  private Option<HoodieCommitMetadata> getCommitMetadataForInstant(HoodieTimeline timeline, String instantTime) throws IOException {
+    List<HoodieInstant> instants = Arrays.asList(
+            new HoodieInstant(false, HoodieTimeline.COMMIT_ACTION, instantTime),
+            new HoodieInstant(false, HoodieTimeline.REPLACE_COMMIT_ACTION, instantTime),
+            new HoodieInstant(false, HoodieTimeline.DELTA_COMMIT_ACTION, instantTime));
+
+    Option<HoodieInstant> hoodieInstant = Option.fromJavaOptional(instants.stream().filter(timeline::containsInstant).findAny());
+
+    if (hoodieInstant.isPresent()) {
+      return Option.of(HoodieCommitMetadata.fromBytes(timeline.getInstantDetails(hoodieInstant.get()).get(),
```

Comment:
I like the earlier implementation where we are actually parsing this as HoodieReplaceCommitMetadata for 'REPLACE_COMMIT'. That allows callers to print additional replace specific information.